From 5615e8f7bbf83f03acacd6d6ba9ff8dc6fdc91df Mon Sep 17 00:00:00 2001
From: "mark.yang" <mark.yang@lge.com>
Date: Thu, 10 Apr 2025 18:49:51 +0900
Subject: [PATCH] fix incompatible type assignment in txring.c

* There is an incompatible type assignment error in txring.c.
  txring.c:200:12: error: incompatible types when assigning to type 'txring_t' {aka 'struct txring_s'} from type 'txring_t *' {aka 'struct txring_s *'}
  200 |     *txp = (txring_t *)safe_malloc(sizeof(txring_t));
      |            ^

  txp is allocated as txring_t *, so I fixed it.

Upstream-Status: Submitted [https://github.com/appneta/tcpreplay/pull/924]
Signed-off-by: mark.yang <mark.yang@lge.com>
---
 src/common/txring.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/txring.c b/src/common/txring.c
index f522893d..b062dac2 100644
--- a/src/common/txring.c
+++ b/src/common/txring.c
@@ -197,7 +197,7 @@ txring_init(int fd, unsigned int mtu)
     txring_t *txp;
 
     /* allocate memory for structure and fill it with different stuff*/
-    *txp = (txring_t *)safe_malloc(sizeof(txring_t));
+    txp = (txring_t *)safe_malloc(sizeof(txring_t));
     txp->treq = (struct tpacket_req *)safe_malloc(sizeof(struct tpacket_req));
 
     txring_mkreq(txp->treq, mtu);
